mindmeld/components/question_answerer.py

```diff
@@ -661,34 +668,55 @@ def build_query(self):
             # 4. matches on synonym if available (exact, word N-gram and character N-gram):
             # for a knowledge base text field the synonym are indexed in a separate field
             # "<field name>$whitelist" if available.
-
-            clause = {
-                "bool": {
-                    "should": [
-                        {
-                            "match": {
-                                self.field: {
-                                    "query": self.value
+            if self.query_type == 'text':
+                clause = {
+                    "bool": {
+                        "should": [
+                            {
+                                "match": {
+                                    self.field: {
+                                        "query": self.value
+                                    }
                                 }
-                            }
-                        },
-                        {
-                            "match": {
-                                self.field + ".normalized_keyword": {
-                                    "query": self.value
+                            },
+                            {
+                                "match": {
+                                    self.field + ".processed_text": {
+                                        "query": self.value
+                                    }
                                 }
                             }
-                        },
-                        {
-                            "match": {
```

Comment:
looks like we don't do matching on character ngrams anymore. guess some of the sub-word matching is taken care of by the matching on word stem in `processed_text` field?

In the past i remember we intentionally did not do stopword removal and stemming to avoid losing information and tried to let the data decide if the specific text span is important in the context. (based on TF/IDF score for example)  But guess from our eval looks like we still have performance gain from them for the unstructured data we are experimenting with?

Reply:
So far we've only handled matching against short phrases or sentences. If we have big passages, intuitively, char-ngrams will always match with most passages and do not help in the ranking process. Like you mentioned, in the evaluation set, we see a steady 7% gain in accuracy without them.